networking/src/p2p/peer.rs

```diff
@@ -582,23 +810,30 @@ async fn begin_process_incoming(
     while net.rx_run.load(Ordering::Acquire) {
         match timeout(READ_TIMEOUT_LONG, rx.read_message::<PeerMessageResponse>()).await {
             Ok(res) => match res {
-                Ok(msg) => {
-                    let should_broadcast_message = net.rx_run.load(Ordering::Acquire);
-                    if should_broadcast_message {
-                        trace!(log, "Message parsed successfully"; "msg" => format!("{:?}", &msg));
-                        event_channel.tell(
-                            Publish {
-                                msg: PeerMessageReceived {
-                                    peer: myself.clone(),
-                                    message: Arc::new(msg),
-                                }
-                                .into(),
-                                topic: NetworkChannelTopic::NetworkEvents.into(),
-                            },
-                            None,
-                        );
+                Ok(msg) => match throttle_quota.lock() {
+                    Ok(ref mut quota) => {
+                        if quota.can_receive(&msg) {
+                            let should_broadcast_message = net.rx_run.load(Ordering::Acquire);
+                            if should_broadcast_message {
+                                trace!(log, "Message parsed successfully"; "msg" => format!("{:?}", &msg));
+                                event_channel.tell(
+                                    Publish {
+                                        msg: PeerMessageReceived {
+                                            peer: myself.clone(),
+                                            message: Arc::new(msg),
+                                        }
+                                        .into(),
+                                        topic: NetworkChannelTopic::NetworkEvents.into(),
+                                    },
+                                    None,
+                                );
+                            }
+                        } else {
+                            warn!(log, "Message is dropped because its receive quota is exceeded"; "msg" => format!("{:?}", msg), "peer_uri" => myself.uri().to_string());
```

Comment:
Same question as above (and same answer I guess). Will this flood the logs if we keep getting a lot of messages and we are over our quota?

Reply:
Thanks for noticing that. Indeed that might be an issue. I'll log the first occurrence and then the total number when doing quota reset.